### PR TITLE
Add gc enable/disable to timer

### PR DESCRIPTION
**Pull request by calebtung**
Allow user to disable garbage collection to reduce overhead when timing. Automatically reenables garbage collection afterwards

### CODE REVIEW

File: nnutils/timing/timer.py [hunk 1]
```diff
@@ -7,19 +8,29 @@
...
+            if disable_garbage_collector:
+                gc.disable()
+                self.is_garbage_collector_disabled = True
             self.start_time = time.time()
+            
 
         return _time_start
 
-    def time_end(self, inference_times: List[float]):
+    def time_end(self, inference_times: List[float]) -> None:
```
Comment (pjjajal) [inline nnutils/timing/timer.py]: The output of this function is None but Callable

**Review by pjjajal — CHANGES_REQUESTED** [review]
You need to make changes to the [`nnutils/__init__.py`](https://github.com/pjjajal/nn-utils/blob/main/nnutils/__init__.py). Namely, you need to change the `time()` method.

File: nnutils/timing/timer.py (hunk 1 above)
Comment (calebtung) [reply nnutils/timing/timer.py]: Just updated!

**Comment by pjjajal** [comment]
@calebtung As we discussed, you still need to make changes to __init__.py.

**Comment by calebtung** [comment]
> @calebtung As we discussed, you still need to make changes to **init**.py.

NO

Ok fine🥚🥚🥚

**Review by pjjajal — APPROVED** [review]
LGTM.

**Pull request merged**